src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += num;
+
+    return adder;
+  }
+
+  return adder;
```

Comment:
Ensure that you have considered all edge cases. Although the task does not require checking the type of the argument, in real-world applications, it might be necessary to validate the input to avoid unexpected behavior.